### PR TITLE
First draft for openSSL-3 security advisory.

### PR DESCRIPTION
**Pull request by garloff**
Signed-off-by: Kurt Garloff <kurt@garloff.de>

### CODE REVIEW

**Comment by garloff** [comment]
We will probably need to finalize this and publish a v1 TODAY.
In order for this to happen, we need to:
* Ensure our review of OSISM managed containers is complete (@berendt)
* Ensure we have reviewed all containers that can be pulled in for the k8s cluster management (@garloff)
* Make sure the wording is appropriate and correct (all)
* Read the publication of the vuln, add links, CVE number, and summarize what this is for our advisory (technically skilled engineers can do this).

File: _i18n/en/_posts/blog/2022-11-01-advisory-spookyssl.md [hunk 1]
```diff
@@ -0,0 +1,131 @@
...
+shared library. Applications that statically linked the vulnerable
+OpenSSL-3.0.x (x<7) library need to be rebuilt using the fixed library
+(3.0.7).
+
+For container-based applications that built containers using affected
+OpenSSL-3 libraries, rebuilding the containers with the fixed version
+and rolling them out freshly is recommended.
+
+We recommend users to also review the workstations and laptops that are
+used by their staff and to install available updates short-term.
```
Comment (fkr) [inline _i18n/en/_posts/blog/2022-11-01-advisory-spookyssl.md:107]: Imho not the scope of this publication.

File: _i18n/en/_posts/blog/2022-11-01-advisory-spookyssl.md [hunk 2]
```diff
@@ -0,0 +1,131 @@
...
+choice amongst technically minded engineers.
+
+## Note
+
+Not yet having done the switch to Ubuntu-22.04 resulted in the exposure
+of providers using the SCS reference implementation to be minimal.
+Similar to the ["dirty pipe"](https://scs.community/security/2022/03/07/advisory-dirty-pipe/)
+case, our choice of technology turned out to be a lucky one.
+We know that there is no guarantee for luck; we do ensure we can react
+quickly and effectively when affected.
```
Comment (fkr) [inline _i18n/en/_posts/blog/2022-11-01-advisory-spookyssl.md:118]: I don't see the rationale for this comment. Imho the advisory should stick to the fact and not elaborate on luck.

**Review by fkr — APPROVED** [review]
I've pushed a few minor wording fixes.
For the other items, I've commented. I'm fine with the advisory as is, however at a few places it has too much "prosa" for my taste and should just stick to the facts.

File: _i18n/en/_posts/blog/2022-11-01-advisory-spookyssl.md [hunk 3]
```diff
@@ -0,0 +1,131 @@
...
+<!--TODO: More details, CVE and links -->
+
+https://github.com/NCSC-NL/OpenSSL-2022
+
+## SCS reference implementation
+
+The host images in the SCS reference implementation come from the
+[OSISM](https://osism.tech) project and are all built on top of Ubuntu 20.04
+LTS (focal). Ubuntu 20.04 uses OpenSSL-1.1.1 and is unaffected by the
+vulnerability. The switch to Ubuntu 22.04 (jammy) which does use OpenSSL-3.0.x
```
[the remark this answers is not in the document]
Comment (garloff) [reply _i18n/en/_posts/blog/2022-11-01-advisory-spookyssl.md:32]: Improved language in 6d0df78bd5c5be93405761bdc14d94bae85a648a.

File: _i18n/en/_posts/blog/2022-11-01-advisory-spookyssl.md (hunk 2 above)
Comment (garloff) [reply _i18n/en/_posts/blog/2022-11-01-advisory-spookyssl.md]: It's gone.

File: _i18n/en/_posts/blog/2022-11-01-advisory-spookyssl.md [hunk 4]
```diff
@@ -0,0 +1,134 @@
...
+---
+
+## The vulnerability
+
+On Tue, 2022-11-01, a vulnerability in the popular OpenSSL library
+became publicly known. It affects OpenSSL-3.0.x versions prior
+to the fixed version 3.0.7; older versions of OpenSSL (1.1.x
+and earlier) are not affected.
+
+<!--TODO: More details, CVE and links -->
```
Comment (berendt) [inline _i18n/en/_posts/blog/2022-11-01-advisory-spookyssl.md:23]: For Ubuntu: https://ubuntu.com/security/CVE-2022-3602

Security advisory: https://mta.openssl.org/pipermail/openssl-announce/2022-November/000243.html

**Review by berendt — COMMENTED** [review]
Priority changed from CRITICAL to HIGH.

File: _i18n/en/_posts/blog/2022-11-01-advisory-spookyssl.md (hunk 4 above)
Comment (garloff) [reply _i18n/en/_posts/blog/2022-11-01-advisory-spookyssl.md]: I have the link to the advisory already, added the Ubuntu link now.

**Comment by garloff** [comment]
I think we're good for a v1, if @berendt confirms the correctness of the information on the OSISM side.
I'd like to push out at 18:30 CET, but @berendt and @fkr can veto this of course.
Otherwise an approving review would be helpful.

**Comment by itrich** [comment]
Please don't forget to create a symlink in the `de` subfolder. Otherwise, the article won't be displayed on the German website.

**Comment by garloff** [comment]
> Please don't forget to create a symlink in the `de` subfolder. Otherwise, the article won't be displayed on the German website.

I would have overlooked this, thanks.
(Maybe we should have functionality to always link an English article when no German version is available?)

Approving review is required in order to merge this.

**Review by fkr — APPROVED** [review]
LGTM. Thanks!

**Comment by garloff** [comment]
Thanks, Felix!

**Pull request merged**